AFQ/recognition/roi.py

```diff
@@ -1,6 +1,9 @@
 import numpy as np
-from scipy.spatial.distance import cdist
-from scipy.ndimage import binary_dilation
+from dipy.core.interpolation import interpolate_scalar_3d
+
+
+def _interp3d(roi, sl):
+    return interpolate_scalar_3d(roi.get_fdata(), np.asarray(sl))[0]
```

Comment:
Should we call `get_fdata` outside of this function? I think that it is called multiple times now, and even if the function doesn't do much, the call itself still takes non-zero time.

Reply:
I think that would be over-optimizaiton. the Nifti object already knows the data is loaded, so it's not loading it again. It's just a function call and thats almost no time. I think the code is a lot cleaner when we pass around these objects that contain both the data and affine, and only unpack them when we need to.